modulemd/v2/modulemd-component-rpm.c

```diff
@@ -291,6 +291,80 @@ modulemd_component_rpm_set_property (GObject *object,
 }
 
 
+gboolean
+modulemd_component_rpm_equals (ModulemdComponentRpm *self,
+                               ModulemdComponentRpm *other)
+{
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT_RPM (self), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT_RPM (other), FALSE);
+
+
+  if (self == other)
+    return FALSE;
+
+
+  GStrv arches_as_strv_self = modulemd_component_rpm_get_arches_as_strv (self);
```

Comment:
This entire section of code can be replaced by using `modulemd_hash_table_sets_are_equal(self->arches, other->arches)`.

Reply:
How are you accessing the "arches" fields for the two ModulemdComponentRpm's using the "->" operator? The documentation doesn't mention any public fields for the objects.

Reply:
This is giving me a warning because modulemd_hash_table_sets_are_equal takes GHashTable * types as arguments while modulemd_component_rpm_get_arches_as_strv returns a GStrv type.

Reply:
> This is giving me a warning because modulemd_hash_table_sets_are_equal takes GHashTable * types as arguments while modulemd_component_rpm_get_arches_as_strv returns a GStrv type.

You don't need to call `modulemd_component_rpm_get_arches_as_strv()` here. This is the internal implementation of this class, so you have direct access to the internal data structure, `self->arches` and `other->arches` which are `GHashTable *` types.